src/Core/Result.php

```diff
@@ -85,20 +85,22 @@ public function __destruct()
      */
     final public function resolve(?float $timeout = null): bool
     {
-        if (!isset($this->response)) {
+        if ($this->isResolved) {
             return true;
         }
 
-        try {
-            if (null !== $timeout && $this->httpClient->stream($this->response, $timeout)->current()->isTimeout()) {
-                return false;
+        foreach ($this->httpClient->stream($this->response, $timeout) as $chunk) {
+            try {
+                if ($chunk->isTimeout()) {
+                    return false;
+                }
+                if ($chunk->isFirst()) {
```

Comment:
Not sure to understand what this change solves.

If we receive a chunk it have to be the first right?

Reply:
in theory, there can be informative chunks before the first one

Reply:
thanks for the hint